legend-engine-query-pure/src/main/java/org/finos/legend/engine/query/pure/api/Execute.java

```diff
@@ -357,10 +415,26 @@ private Response executeAsMiddleTierPlan(PlanExecutor planExecutor, MutableList<
                 )
         );
 
+        MutableMap<String, Result> vars = Maps.mutable.empty();
+        List<Variable> functionParameters = ((SingleExecutionPlan) plan).rootExecutionNode.executionNodes.stream().filter(node -> node instanceof FunctionParametersValidationNode).map(executionNode -> ((FunctionParametersValidationNode) executionNode).functionParameters).flatMap(Collection::stream).collect(Collectors.toList());
+        Multiplicity paraMultiplicity = new Multiplicity();
+        for (Map.Entry<String, ?> e : parameters.entrySet())
+        {
+            paraMultiplicity = getFunctionParameterMultiplicity(functionParameters, paraMultiplicity, e);
```

Comment:
lets add `parameterValue = parameterToValue.getValue()` before this and use it below to make code more readible